client/post-editor/editor-image-selector/dropzone.jsx

```diff
@@ -0,0 +1,101 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { head, uniqueId } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DropZone from 'components/drop-zone';
+import MediaActions from 'lib/media/actions';
+import MediaStore from 'lib/media/store';
+import { filterItemsByMimePrefix, isItemBeingUploaded } from 'lib/media/utils';
+import ImageSelectorDropZoneIcon from './dropzone-icon';
+
+import { receiveMedia, deleteMedia } from 'state/media/actions';
+import { getSelectedSiteId, getSelectedSite } from 'state/ui/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+class ImageSelectorDropZone extends Component {
+	onFilesDrop = files => {
+		/**
+		 * Filter files for `image` media prefix and return the first image.
+		 *
+		 * At the moment we ignore all the other images that were dragged onto the DropZone
+		 */
+		const droppedImage = head( filterItemsByMimePrefix( files, 'image' ) );
+
+		if ( ! droppedImage ) {
+			return false;
+		}
+
+		const transientMediaId = uniqueId( 'image-selector' );
+		const { siteId, site } = this.props;
+
+		const handleImageSelectorUpload = () => {
+			const media = MediaStore.get( siteId, transientMediaId );
+			const isUploadInProgress = media && isItemBeingUploaded( media );
+			const isFailedUpload = ! media;
+
+			if ( isFailedUpload ) {
+				this.props.deleteMedia( siteId, transientMediaId );
+			} else {
+				this.props.receiveMedia( siteId, media );
+			}
+
+			/**
+			 * File upload finished. No need to listen for changes anymore.
+			 */
+			if ( ! isUploadInProgress ) {
+				MediaStore.off( 'change', handleImageSelectorUpload );
+
+				// Successful image upload.
+				if ( media ) {
+					this.props.recordTracksEvent( 'calypso_editor_image_selector_upload', {
```

Comment:
Since this component might be used elsewhere, wondering about the naming of this track event.

Reply:
You're right.  Since it's going to be used elsewhere I think the best way to handle this is to remove it entirely and handle it in the props when the dropped image is passed back.